### ISSUE TITLE
Add support for other authentication mechanisms for NATS and STAN

### ISSUE THREAD

**Issue by ldclakmal** [ballerina-platform/ballerina-standard-library#1467]
**Description:**
Currently, Ballerina NATS and STAN does support for Token based authentication and Username/Password credentials based authentication. Need to add support for other authentication mechanisms defined in https://docs.nats.io/nats-server/configuration/securing_nats/auth_intro

**Comment by dwaynebradley** [comment]
NATS is a core component in just about every project I work on. Not having support for NKeys and JWTs prevents me from using Ballerina in any meaningful way on these projects. This functionality is already in the underlying Java implementation that Ballerina uses so hopefully it wouldn't be too difficult to pick up.

Also, STAN should be deprecated at this point. It is now considered "legacy": https://docs.nats.io/legacy/stan

JetStream is the path forward for persistence and it built directly into `nats-server`. This brings in a whole new list of functionality that Ballerina should be able to support as well:

- Steams
- Consumers
- Key/Value storage
- Object storage

See here: https://docs.nats.io/nats-concepts/jetstream

**Comment by ldclakmal** [comment]
> NATS is a core component in just about every project I work on. Not having support for NKeys and JWTs prevents me from using Ballerina in any meaningful way on these projects. This functionality is already in the underlying Java implementation that Ballerina uses so hopefully it wouldn't be too difficult to pick up.

@dwaynebradley Thanks for the comment! We will looking to this and check on providing the support of NKeys and JWT auth mechanisms with a new release of the NATS module.

**Comment by shafreenAnfar** [comment]
@dwaynebradley yes, STAN is deprecated and JetStream is the new alternative. We are yet to work on JetStream. I have created an issue to track this requirement.
